### PR TITLE
Prioritize running custom checks

### PR DESCRIPTION
**Pull request by robeden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Logic has been adjusted so that custom checks are always run (unless specifically skipped), regardless of whether or not there are `--check` flags.

A big part of this task was adding unit tests (`tests/common/test_runner_filter.py`, `tests/kubernetes/test_base_registry.py`) to ensure the behavior is well understood both before and after changes. In particular, the logic for the Kubernetes registry is quite different from other registries (i.e., run loops) and was quite complex.

~~⚠️ **HEADS UP** ⚠️  - There are two `TODO` questions in `tests/kubernetes/test_base_registry.py` which should be looked at and either resolved or removed. The tests and current behavior match the logic before I made changes, but the results seem backwards to me.~~ Confirmed correct behavior.

Fixes: #466

### CODE REVIEW

File: checkov/kubernetes/base_registry.py [hunk 1]
```diff
@@ -11,74 +11,75 @@ def extract_entity_details(self, entity):
...
                 self.logger.debug("Running check: {} on file {}".format(check.name, scanned_file))
+
                 result = check.run(scanned_file=scanned_file, entity_configuration=entity_configuration,
                                    entity_name=entity_type, entity_type=entity_type, skip_info=skip_info)
-
                 results[check] = result
         return results
+
+    @staticmethod
+    def _should_run_scan(check_id, entity_configuration, runner_filter):
```
Comment (robeden) [inline checkov/kubernetes/base_registry.py:33]: NOTE: This was broken out to allow unit testing.

File: checkov/runner_filter.py [hunk 2]
```diff
@@ -18,3 +19,18 @@ def __init__(self, framework='all', checks=None, skip_checks=None):
         else:
             self.skip_checks = skip_checks
         self.framework = framework
+
+    def notify_external_check(self, check_id):
+        self.external_check_ids.add(check_id)
+
+    def should_run_check(self, check_id):
+        if check_id in self.external_check_ids:
+            pass        # enabled unless kipped
```
Comment (KevinHock) [inline checkov/runner_filter.py:28]: Nit: s/kipped/skipped/

Comment (robeden) [reply checkov/runner_filter.py]: 👍

**Comment by robeden** [comment]
Great, thanks @jonjozwiak. I’ll remove the todos.

**Pull request merged**